libcontainer/rootfs_linux.go

```diff
@@ -400,7 +400,29 @@ func mountToRootfs(m *configs.Mount, c *mountConfig) error {
 			return err
 		}
 		// Selinux kernels do not support labeling of /proc or /sys
-		return mountPropagate(m, rootfs, "", nil)
+		err = mountPropagate(m, rootfs, "", nil)
+		if err != nil && m.Device == "sysfs" {
+			/* If we are running in a user namespace, just bind mount /sys if creating
+			   sysfs failed.  */
+			if !userns.RunningInUserNS() {
+				return err
+			}
+			m.Source, m.Device = "/sys", "bind"
+			m.Flags = unix.MS_BIND | unix.MS_REC | unix.MS_NODEV | unix.MS_NOSUID | unix.MS_NOEXEC
+			if err = utils.WithProcfd(rootfs, m.Destination, func(procfd string) error {
+				return mount(m.Source, m.Destination, procfd, m.Device, uintptr(m.Flags), label.FormatMountLabel(m.Data, mountLabel))
```

Comment:
For a bind mount, fstype and data arguments are ignored, no need to set or pass those.

Reply:
yes, you are right, thanks